src/Common/src/TypeSystem/Common/TypeSystemHelpers.cs

```diff
@@ -214,5 +215,26 @@ static public MethodDesc TryResolveConstraintMethodApprox(this MetadataType cons
 
             return method;
         }
+
+        /// <summary>
+        /// Retrieves the namespace qualified name of a <see cref="MetadataType"/>.
+        /// </summary>
+        public static string GetFullName(this MetadataType metadataType)
+        {
+            string ns = metadataType.Namespace;
+            bool hasNs = ns.Length > 0;
```

Comment:
`return (ns.Length > 0) ? String.Concat(ns, ".", name) : name;` is simpler and more efficient.